contracts/game/src/lib.cairo

```diff
@@ -1083,6 +1130,27 @@ mod Game {
         _save_adventurer_no_boosts(ref self, adventurer, adventurer_id);
     }
 
+    fn _slay_invalid_adventurer(ref self: ContractState, adventurer_id: felt252) {
+        // unpack adventurer from storage (no need for stat boosts)
+        let mut adventurer = _load_adventurer_no_boosts(@self, adventurer_id);
+
+        // assert adventurer is not already dead
+        _assert_not_dead(adventurer);
+
+        // assert adventurer is idle
+        _assert_is_idle(@self, adventurer, adventurer_id);
```

Comment:
Why do we need to wait for idle, can we stop the invalid adventurer from getting a good score and collecting any payment?